contracts/StablecoinConverter.sol

```diff
@@ -317,17 +327,20 @@ contract StablecoinConverter is EpochTokenLocker {
             subtractBalance(
                 previousSolution.solutionSubmitter,
                 tokenIdToAddressMap(0),
-                uint(previousSolution.currentFeeCollected) / 2
+                previousSolution.currentFeeReward
             );
         }
     }
 
-    function checkAndOverrideObjectiveValue(int256 fee, uint32 batchIndex) private {
+    function checkAndOverrideObjectiveValue(uint256 newObjectiveValue, uint32 batchIndex) private {
         if (previousSolution.batchId < batchIndex) {
-            previousSolution.currentFeeCollected = 0;
+            previousSolution.currentObjectiveValue = 0;
         }
-        require(fee > previousSolution.currentFeeCollected, "Solution does not generate a higher fee than a previous solution");
-        previousSolution.currentFeeCollected = fee;
+        require(
```

Comment:
Can this now become

```
require(newObjectiveValue > getCurrentObjectiveValue())
```

and get rid of the if above to avoid setting currentObjectiveValue twice?

On another note, if we go back to utility, do we still want to allow solutions that have 0 utility if no solution has previously been submitted?